src/edu/csus/ecs/pc2/ui/SubmissionSolutionList.java

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.ui;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import edu.csus.ecs.pc2.imports.ccs.IContestLoader;
+
+/**
+ * List of judge's solutions
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class SubmissionSolutionList extends ArrayList<SubmissionSampleLocation> {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 6231245379952020474L;
+	
+	public SubmissionSolutionList() {
+		super();
+	}
+	
+	public SubmissionSolutionList(File cdpPath) {
+		super();
+		loadCDPSampleSubmissionList(cdpPath);
+	}
+	
+
+    /**
+     * get all directories and child directories (recurses).
+     * 
+     * @param directory
+     * @return list of directory names
+     */
+	// TODO REFACTOR move to FileUtilities.getAllDirectoryEntries
+    public static List<String> getAllDirectoryEntries(String directory) {
```

Comment:
As noted in the above comment, this method doesn't belong in class `SubmissionSolutionList`; it appears to belong in `FileUtilities`.